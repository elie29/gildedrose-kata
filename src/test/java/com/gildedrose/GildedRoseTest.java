package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.approvaltests.reporters.ClipboardReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

@UseReporter(ClipboardReporter.class)
class GildedRoseTest {

   @Test
   void testUpdateQuality() {
      CombinationApprovals.verifyAllCombinations(
         this::doUpdate,
         new String[] { "foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros" },
         new Integer[] { -1, 0, 5, 6, 7, 10, 11, 12 },
         new Integer[] { -1, 0, 1, 2, 48, 49, 50, 51 }
      );
   }

   private String doUpdate(String name, int sellIn, int quality) {
      Item[] items = new Item[] { ItemFactory.of(name, sellIn, quality) };
      GildedRose app = new GildedRose(items);
      app.updateQuality();

      return Arrays.toString(items);
   }

}
