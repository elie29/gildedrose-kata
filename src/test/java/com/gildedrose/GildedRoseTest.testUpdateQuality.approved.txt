[foo, -1, -1] => [foo, -2, -1] 
[foo, -1, 0] => [foo, -2, 0] 
[foo, -1, 1] => [foo, -2, 0] 
[foo, -1, 2] => [foo, -2, 0] 
[foo, -1, 48] => [foo, -2, 46] 
[foo, -1, 49] => [foo, -2, 47] 
[foo, -1, 50] => [foo, -2, 48] 
[foo, -1, 51] => [foo, -2, 49] 
[foo, 0, -1] => [foo, -1, -1] 
[foo, 0, 0] => [foo, -1, 0] 
[foo, 0, 1] => [foo, -1, 0] 
[foo, 0, 2] => [foo, -1, 0] 
[foo, 0, 48] => [foo, -1, 46] 
[foo, 0, 49] => [foo, -1, 47] 
[foo, 0, 50] => [foo, -1, 48] 
[foo, 0, 51] => [foo, -1, 49] 
[foo, 5, -1] => [foo, 4, -1] 
[foo, 5, 0] => [foo, 4, 0] 
[foo, 5, 1] => [foo, 4, 0] 
[foo, 5, 2] => [foo, 4, 1] 
[foo, 5, 48] => [foo, 4, 47] 
[foo, 5, 49] => [foo, 4, 48] 
[foo, 5, 50] => [foo, 4, 49] 
[foo, 5, 51] => [foo, 4, 50] 
[foo, 6, -1] => [foo, 5, -1] 
[foo, 6, 0] => [foo, 5, 0] 
[foo, 6, 1] => [foo, 5, 0] 
[foo, 6, 2] => [foo, 5, 1] 
[foo, 6, 48] => [foo, 5, 47] 
[foo, 6, 49] => [foo, 5, 48] 
[foo, 6, 50] => [foo, 5, 49] 
[foo, 6, 51] => [foo, 5, 50] 
[foo, 7, -1] => [foo, 6, -1] 
[foo, 7, 0] => [foo, 6, 0] 
[foo, 7, 1] => [foo, 6, 0] 
[foo, 7, 2] => [foo, 6, 1] 
[foo, 7, 48] => [foo, 6, 47] 
[foo, 7, 49] => [foo, 6, 48] 
[foo, 7, 50] => [foo, 6, 49] 
[foo, 7, 51] => [foo, 6, 50] 
[foo, 10, -1] => [foo, 9, -1] 
[foo, 10, 0] => [foo, 9, 0] 
[foo, 10, 1] => [foo, 9, 0] 
[foo, 10, 2] => [foo, 9, 1] 
[foo, 10, 48] => [foo, 9, 47] 
[foo, 10, 49] => [foo, 9, 48] 
[foo, 10, 50] => [foo, 9, 49] 
[foo, 10, 51] => [foo, 9, 50] 
[foo, 11, -1] => [foo, 10, -1] 
[foo, 11, 0] => [foo, 10, 0] 
[foo, 11, 1] => [foo, 10, 0] 
[foo, 11, 2] => [foo, 10, 1] 
[foo, 11, 48] => [foo, 10, 47] 
[foo, 11, 49] => [foo, 10, 48] 
[foo, 11, 50] => [foo, 10, 49] 
[foo, 11, 51] => [foo, 10, 50] 
[foo, 12, -1] => [foo, 11, -1] 
[foo, 12, 0] => [foo, 11, 0] 
[foo, 12, 1] => [foo, 11, 0] 
[foo, 12, 2] => [foo, 11, 1] 
[foo, 12, 48] => [foo, 11, 47] 
[foo, 12, 49] => [foo, 11, 48] 
[foo, 12, 50] => [foo, 11, 49] 
[foo, 12, 51] => [foo, 11, 50] 
[Aged Brie, -1, -1] => [Aged Brie, -2, 1] 
[Aged Brie, -1, 0] => [Aged Brie, -2, 2] 
[Aged Brie, -1, 1] => [Aged Brie, -2, 3] 
[Aged Brie, -1, 2] => [Aged Brie, -2, 4] 
[Aged Brie, -1, 48] => [Aged Brie, -2, 50] 
[Aged Brie, -1, 49] => [Aged Brie, -2, 50] 
[Aged Brie, -1, 50] => [Aged Brie, -2, 50] 
[Aged Brie, -1, 51] => [Aged Brie, -2, 51] 
[Aged Brie, 0, -1] => [Aged Brie, -1, 1] 
[Aged Brie, 0, 0] => [Aged Brie, -1, 2] 
[Aged Brie, 0, 1] => [Aged Brie, -1, 3] 
[Aged Brie, 0, 2] => [Aged Brie, -1, 4] 
[Aged Brie, 0, 48] => [Aged Brie, -1, 50] 
[Aged Brie, 0, 49] => [Aged Brie, -1, 50] 
[Aged Brie, 0, 50] => [Aged Brie, -1, 50] 
[Aged Brie, 0, 51] => [Aged Brie, -1, 51] 
[Aged Brie, 5, -1] => [Aged Brie, 4, 0] 
[Aged Brie, 5, 0] => [Aged Brie, 4, 1] 
[Aged Brie, 5, 1] => [Aged Brie, 4, 2] 
[Aged Brie, 5, 2] => [Aged Brie, 4, 3] 
[Aged Brie, 5, 48] => [Aged Brie, 4, 49] 
[Aged Brie, 5, 49] => [Aged Brie, 4, 50] 
[Aged Brie, 5, 50] => [Aged Brie, 4, 50] 
[Aged Brie, 5, 51] => [Aged Brie, 4, 51] 
[Aged Brie, 6, -1] => [Aged Brie, 5, 0] 
[Aged Brie, 6, 0] => [Aged Brie, 5, 1] 
[Aged Brie, 6, 1] => [Aged Brie, 5, 2] 
[Aged Brie, 6, 2] => [Aged Brie, 5, 3] 
[Aged Brie, 6, 48] => [Aged Brie, 5, 49] 
[Aged Brie, 6, 49] => [Aged Brie, 5, 50] 
[Aged Brie, 6, 50] => [Aged Brie, 5, 50] 
[Aged Brie, 6, 51] => [Aged Brie, 5, 51] 
[Aged Brie, 7, -1] => [Aged Brie, 6, 0] 
[Aged Brie, 7, 0] => [Aged Brie, 6, 1] 
[Aged Brie, 7, 1] => [Aged Brie, 6, 2] 
[Aged Brie, 7, 2] => [Aged Brie, 6, 3] 
[Aged Brie, 7, 48] => [Aged Brie, 6, 49] 
[Aged Brie, 7, 49] => [Aged Brie, 6, 50] 
[Aged Brie, 7, 50] => [Aged Brie, 6, 50] 
[Aged Brie, 7, 51] => [Aged Brie, 6, 51] 
[Aged Brie, 10, -1] => [Aged Brie, 9, 0] 
[Aged Brie, 10, 0] => [Aged Brie, 9, 1] 
[Aged Brie, 10, 1] => [Aged Brie, 9, 2] 
[Aged Brie, 10, 2] => [Aged Brie, 9, 3] 
[Aged Brie, 10, 48] => [Aged Brie, 9, 49] 
[Aged Brie, 10, 49] => [Aged Brie, 9, 50] 
[Aged Brie, 10, 50] => [Aged Brie, 9, 50] 
[Aged Brie, 10, 51] => [Aged Brie, 9, 51] 
[Aged Brie, 11, -1] => [Aged Brie, 10, 0] 
[Aged Brie, 11, 0] => [Aged Brie, 10, 1] 
[Aged Brie, 11, 1] => [Aged Brie, 10, 2] 
[Aged Brie, 11, 2] => [Aged Brie, 10, 3] 
[Aged Brie, 11, 48] => [Aged Brie, 10, 49] 
[Aged Brie, 11, 49] => [Aged Brie, 10, 50] 
[Aged Brie, 11, 50] => [Aged Brie, 10, 50] 
[Aged Brie, 11, 51] => [Aged Brie, 10, 51] 
[Aged Brie, 12, -1] => [Aged Brie, 11, 0] 
[Aged Brie, 12, 0] => [Aged Brie, 11, 1] 
[Aged Brie, 12, 1] => [Aged Brie, 11, 2] 
[Aged Brie, 12, 2] => [Aged Brie, 11, 3] 
[Aged Brie, 12, 48] => [Aged Brie, 11, 49] 
[Aged Brie, 12, 49] => [Aged Brie, 11, 50] 
[Aged Brie, 12, 50] => [Aged Brie, 11, 50] 
[Aged Brie, 12, 51] => [Aged Brie, 11, 51] 
[Backstage passes to a TAFKAL80ETC concert, -1, -1] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 1] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 2] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 48] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, -1, 51] => [Backstage passes to a TAFKAL80ETC concert, -2, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, -1] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 48] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] => [Backstage passes to a TAFKAL80ETC concert, -1, 0] 
[Backstage passes to a TAFKAL80ETC concert, 5, -1] => [Backstage passes to a TAFKAL80ETC concert, 4, 2] 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => [Backstage passes to a TAFKAL80ETC concert, 4, 3] 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] => [Backstage passes to a TAFKAL80ETC concert, 4, 4] 
[Backstage passes to a TAFKAL80ETC concert, 5, 2] => [Backstage passes to a TAFKAL80ETC concert, 4, 5] 
[Backstage passes to a TAFKAL80ETC concert, 5, 48] => [Backstage passes to a TAFKAL80ETC concert, 4, 50] 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] => [Backstage passes to a TAFKAL80ETC concert, 4, 50] 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => [Backstage passes to a TAFKAL80ETC concert, 4, 50] 
[Backstage passes to a TAFKAL80ETC concert, 5, 51] => [Backstage passes to a TAFKAL80ETC concert, 4, 51] 
[Backstage passes to a TAFKAL80ETC concert, 6, -1] => [Backstage passes to a TAFKAL80ETC concert, 5, 1] 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] => [Backstage passes to a TAFKAL80ETC concert, 5, 2] 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] => [Backstage passes to a TAFKAL80ETC concert, 5, 3] 
[Backstage passes to a TAFKAL80ETC concert, 6, 2] => [Backstage passes to a TAFKAL80ETC concert, 5, 4] 
[Backstage passes to a TAFKAL80ETC concert, 6, 48] => [Backstage passes to a TAFKAL80ETC concert, 5, 50] 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] => [Backstage passes to a TAFKAL80ETC concert, 5, 50] 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] => [Backstage passes to a TAFKAL80ETC concert, 5, 50] 
[Backstage passes to a TAFKAL80ETC concert, 6, 51] => [Backstage passes to a TAFKAL80ETC concert, 5, 51] 
[Backstage passes to a TAFKAL80ETC concert, 7, -1] => [Backstage passes to a TAFKAL80ETC concert, 6, 1] 
[Backstage passes to a TAFKAL80ETC concert, 7, 0] => [Backstage passes to a TAFKAL80ETC concert, 6, 2] 
[Backstage passes to a TAFKAL80ETC concert, 7, 1] => [Backstage passes to a TAFKAL80ETC concert, 6, 3] 
[Backstage passes to a TAFKAL80ETC concert, 7, 2] => [Backstage passes to a TAFKAL80ETC concert, 6, 4] 
[Backstage passes to a TAFKAL80ETC concert, 7, 48] => [Backstage passes to a TAFKAL80ETC concert, 6, 50] 
[Backstage passes to a TAFKAL80ETC concert, 7, 49] => [Backstage passes to a TAFKAL80ETC concert, 6, 50] 
[Backstage passes to a TAFKAL80ETC concert, 7, 50] => [Backstage passes to a TAFKAL80ETC concert, 6, 50] 
[Backstage passes to a TAFKAL80ETC concert, 7, 51] => [Backstage passes to a TAFKAL80ETC concert, 6, 51] 
[Backstage passes to a TAFKAL80ETC concert, 10, -1] => [Backstage passes to a TAFKAL80ETC concert, 9, 1] 
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => [Backstage passes to a TAFKAL80ETC concert, 9, 2] 
[Backstage passes to a TAFKAL80ETC concert, 10, 1] => [Backstage passes to a TAFKAL80ETC concert, 9, 3] 
[Backstage passes to a TAFKAL80ETC concert, 10, 2] => [Backstage passes to a TAFKAL80ETC concert, 9, 4] 
[Backstage passes to a TAFKAL80ETC concert, 10, 48] => [Backstage passes to a TAFKAL80ETC concert, 9, 50] 
[Backstage passes to a TAFKAL80ETC concert, 10, 49] => [Backstage passes to a TAFKAL80ETC concert, 9, 50] 
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => [Backstage passes to a TAFKAL80ETC concert, 9, 50] 
[Backstage passes to a TAFKAL80ETC concert, 10, 51] => [Backstage passes to a TAFKAL80ETC concert, 9, 51] 
[Backstage passes to a TAFKAL80ETC concert, 11, -1] => [Backstage passes to a TAFKAL80ETC concert, 10, 0] 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] => [Backstage passes to a TAFKAL80ETC concert, 10, 1] 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] => [Backstage passes to a TAFKAL80ETC concert, 10, 2] 
[Backstage passes to a TAFKAL80ETC concert, 11, 2] => [Backstage passes to a TAFKAL80ETC concert, 10, 3] 
[Backstage passes to a TAFKAL80ETC concert, 11, 48] => [Backstage passes to a TAFKAL80ETC concert, 10, 49] 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] => [Backstage passes to a TAFKAL80ETC concert, 10, 50] 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] => [Backstage passes to a TAFKAL80ETC concert, 10, 50] 
[Backstage passes to a TAFKAL80ETC concert, 11, 51] => [Backstage passes to a TAFKAL80ETC concert, 10, 51] 
[Backstage passes to a TAFKAL80ETC concert, 12, -1] => [Backstage passes to a TAFKAL80ETC concert, 11, 0] 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => [Backstage passes to a TAFKAL80ETC concert, 11, 1] 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] => [Backstage passes to a TAFKAL80ETC concert, 11, 2] 
[Backstage passes to a TAFKAL80ETC concert, 12, 2] => [Backstage passes to a TAFKAL80ETC concert, 11, 3] 
[Backstage passes to a TAFKAL80ETC concert, 12, 48] => [Backstage passes to a TAFKAL80ETC concert, 11, 49] 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => [Backstage passes to a TAFKAL80ETC concert, 11, 50] 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => [Backstage passes to a TAFKAL80ETC concert, 11, 50] 
[Backstage passes to a TAFKAL80ETC concert, 12, 51] => [Backstage passes to a TAFKAL80ETC concert, 11, 51] 
[Sulfuras, Hand of Ragnaros, -1, -1] => [Sulfuras, Hand of Ragnaros, -1, -1] 
[Sulfuras, Hand of Ragnaros, -1, 0] => [Sulfuras, Hand of Ragnaros, -1, 0] 
[Sulfuras, Hand of Ragnaros, -1, 1] => [Sulfuras, Hand of Ragnaros, -1, 1] 
[Sulfuras, Hand of Ragnaros, -1, 2] => [Sulfuras, Hand of Ragnaros, -1, 2] 
[Sulfuras, Hand of Ragnaros, -1, 48] => [Sulfuras, Hand of Ragnaros, -1, 48] 
[Sulfuras, Hand of Ragnaros, -1, 49] => [Sulfuras, Hand of Ragnaros, -1, 49] 
[Sulfuras, Hand of Ragnaros, -1, 50] => [Sulfuras, Hand of Ragnaros, -1, 50] 
[Sulfuras, Hand of Ragnaros, -1, 51] => [Sulfuras, Hand of Ragnaros, -1, 51] 
[Sulfuras, Hand of Ragnaros, 0, -1] => [Sulfuras, Hand of Ragnaros, 0, -1] 
[Sulfuras, Hand of Ragnaros, 0, 0] => [Sulfuras, Hand of Ragnaros, 0, 0] 
[Sulfuras, Hand of Ragnaros, 0, 1] => [Sulfuras, Hand of Ragnaros, 0, 1] 
[Sulfuras, Hand of Ragnaros, 0, 2] => [Sulfuras, Hand of Ragnaros, 0, 2] 
[Sulfuras, Hand of Ragnaros, 0, 48] => [Sulfuras, Hand of Ragnaros, 0, 48] 
[Sulfuras, Hand of Ragnaros, 0, 49] => [Sulfuras, Hand of Ragnaros, 0, 49] 
[Sulfuras, Hand of Ragnaros, 0, 50] => [Sulfuras, Hand of Ragnaros, 0, 50] 
[Sulfuras, Hand of Ragnaros, 0, 51] => [Sulfuras, Hand of Ragnaros, 0, 51] 
[Sulfuras, Hand of Ragnaros, 5, -1] => [Sulfuras, Hand of Ragnaros, 5, -1] 
[Sulfuras, Hand of Ragnaros, 5, 0] => [Sulfuras, Hand of Ragnaros, 5, 0] 
[Sulfuras, Hand of Ragnaros, 5, 1] => [Sulfuras, Hand of Ragnaros, 5, 1] 
[Sulfuras, Hand of Ragnaros, 5, 2] => [Sulfuras, Hand of Ragnaros, 5, 2] 
[Sulfuras, Hand of Ragnaros, 5, 48] => [Sulfuras, Hand of Ragnaros, 5, 48] 
[Sulfuras, Hand of Ragnaros, 5, 49] => [Sulfuras, Hand of Ragnaros, 5, 49] 
[Sulfuras, Hand of Ragnaros, 5, 50] => [Sulfuras, Hand of Ragnaros, 5, 50] 
[Sulfuras, Hand of Ragnaros, 5, 51] => [Sulfuras, Hand of Ragnaros, 5, 51] 
[Sulfuras, Hand of Ragnaros, 6, -1] => [Sulfuras, Hand of Ragnaros, 6, -1] 
[Sulfuras, Hand of Ragnaros, 6, 0] => [Sulfuras, Hand of Ragnaros, 6, 0] 
[Sulfuras, Hand of Ragnaros, 6, 1] => [Sulfuras, Hand of Ragnaros, 6, 1] 
[Sulfuras, Hand of Ragnaros, 6, 2] => [Sulfuras, Hand of Ragnaros, 6, 2] 
[Sulfuras, Hand of Ragnaros, 6, 48] => [Sulfuras, Hand of Ragnaros, 6, 48] 
[Sulfuras, Hand of Ragnaros, 6, 49] => [Sulfuras, Hand of Ragnaros, 6, 49] 
[Sulfuras, Hand of Ragnaros, 6, 50] => [Sulfuras, Hand of Ragnaros, 6, 50] 
[Sulfuras, Hand of Ragnaros, 6, 51] => [Sulfuras, Hand of Ragnaros, 6, 51] 
[Sulfuras, Hand of Ragnaros, 7, -1] => [Sulfuras, Hand of Ragnaros, 7, -1] 
[Sulfuras, Hand of Ragnaros, 7, 0] => [Sulfuras, Hand of Ragnaros, 7, 0] 
[Sulfuras, Hand of Ragnaros, 7, 1] => [Sulfuras, Hand of Ragnaros, 7, 1] 
[Sulfuras, Hand of Ragnaros, 7, 2] => [Sulfuras, Hand of Ragnaros, 7, 2] 
[Sulfuras, Hand of Ragnaros, 7, 48] => [Sulfuras, Hand of Ragnaros, 7, 48] 
[Sulfuras, Hand of Ragnaros, 7, 49] => [Sulfuras, Hand of Ragnaros, 7, 49] 
[Sulfuras, Hand of Ragnaros, 7, 50] => [Sulfuras, Hand of Ragnaros, 7, 50] 
[Sulfuras, Hand of Ragnaros, 7, 51] => [Sulfuras, Hand of Ragnaros, 7, 51] 
[Sulfuras, Hand of Ragnaros, 10, -1] => [Sulfuras, Hand of Ragnaros, 10, -1] 
[Sulfuras, Hand of Ragnaros, 10, 0] => [Sulfuras, Hand of Ragnaros, 10, 0] 
[Sulfuras, Hand of Ragnaros, 10, 1] => [Sulfuras, Hand of Ragnaros, 10, 1] 
[Sulfuras, Hand of Ragnaros, 10, 2] => [Sulfuras, Hand of Ragnaros, 10, 2] 
[Sulfuras, Hand of Ragnaros, 10, 48] => [Sulfuras, Hand of Ragnaros, 10, 48] 
[Sulfuras, Hand of Ragnaros, 10, 49] => [Sulfuras, Hand of Ragnaros, 10, 49] 
[Sulfuras, Hand of Ragnaros, 10, 50] => [Sulfuras, Hand of Ragnaros, 10, 50] 
[Sulfuras, Hand of Ragnaros, 10, 51] => [Sulfuras, Hand of Ragnaros, 10, 51] 
[Sulfuras, Hand of Ragnaros, 11, -1] => [Sulfuras, Hand of Ragnaros, 11, -1] 
[Sulfuras, Hand of Ragnaros, 11, 0] => [Sulfuras, Hand of Ragnaros, 11, 0] 
[Sulfuras, Hand of Ragnaros, 11, 1] => [Sulfuras, Hand of Ragnaros, 11, 1] 
[Sulfuras, Hand of Ragnaros, 11, 2] => [Sulfuras, Hand of Ragnaros, 11, 2] 
[Sulfuras, Hand of Ragnaros, 11, 48] => [Sulfuras, Hand of Ragnaros, 11, 48] 
[Sulfuras, Hand of Ragnaros, 11, 49] => [Sulfuras, Hand of Ragnaros, 11, 49] 
[Sulfuras, Hand of Ragnaros, 11, 50] => [Sulfuras, Hand of Ragnaros, 11, 50] 
[Sulfuras, Hand of Ragnaros, 11, 51] => [Sulfuras, Hand of Ragnaros, 11, 51] 
[Sulfuras, Hand of Ragnaros, 12, -1] => [Sulfuras, Hand of Ragnaros, 12, -1] 
[Sulfuras, Hand of Ragnaros, 12, 0] => [Sulfuras, Hand of Ragnaros, 12, 0] 
[Sulfuras, Hand of Ragnaros, 12, 1] => [Sulfuras, Hand of Ragnaros, 12, 1] 
[Sulfuras, Hand of Ragnaros, 12, 2] => [Sulfuras, Hand of Ragnaros, 12, 2] 
[Sulfuras, Hand of Ragnaros, 12, 48] => [Sulfuras, Hand of Ragnaros, 12, 48] 
[Sulfuras, Hand of Ragnaros, 12, 49] => [Sulfuras, Hand of Ragnaros, 12, 49] 
[Sulfuras, Hand of Ragnaros, 12, 50] => [Sulfuras, Hand of Ragnaros, 12, 50] 
[Sulfuras, Hand of Ragnaros, 12, 51] => [Sulfuras, Hand of Ragnaros, 12, 51] 
